client/additional-methods-setup/index.js

```diff
@@ -0,0 +1,41 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies.
+ */
+import Page from 'components/page';
+import MethodSelector from './methods-selector';
+import UpePreviewMethodSelector from './upe-preview-methods-selector';
+import WcPayUpeContextProvider from '../settings/wcpay-upe-toggle/provider';
+import WCPaySettingsContext from '../settings/wcpay-settings-context';
+
+const AdditionalMethodsPage = () => {
+	const {
+		isUpeSettingsPreviewEnabled,
+		isUpeEnabled,
+	} = window.wcpaySettings.additionalMethodsSetup;
+
+	const additionalMethodsContent = isUpeSettingsPreviewEnabled ? (
+		<>
+			<WcPayUpeContextProvider defaultIsUpeEnabled={ isUpeEnabled }>
+				<UpePreviewMethodSelector />
+			</WcPayUpeContextProvider>
+		</>
+	) : (
+		<MethodSelector />
```

Comment:
Is there a situation where we show this page with UPE off, making `<MethodSelector />` appear?

Reply:
I didn't find any "natural" flow where that'd happen. The wp_option `_wcpay_feature_upe_settings_preview`is true by default and isn't set nor updated in our code, which is the option that defines the `isUpeSettingsPreviewEnabled` variable here.

The reason why I added this conditional is to keep the behavior from the component I removed, [here](https://github.com/Automattic/woocommerce-payments/pull/3134/files/9e9a328e10e465ad92df0ad2f20c1393a5f8d01b#diff-64da8dc3c200ed8ee47daef3203920aa6d1554e509a9eda184c959844b90991bL57). By the time I checked, even when you disabled UPE preview, this module wasn't saving due to an error. I'll check again but if this is the case, we might be good removing it. We could check with Heisenberg if so.

Reply:
Adding some more context here. I don't see paths that would take to this page when UPE preview is disabled after this PR is merged. And when you go there directly, UPE methods are displayed as available. Maybe we better remove `<MethodSelector />` and prevent this screen from being displayed if UPE preview is disabled? 

Asking here because we're on it, but we could probably address it in a separate PR.

Reply:
When we first developed UPE and introduce a new settings redesign page, we [introduced](https://github.com/Automattic/woocommerce-payments/issues/2154) this feature flag `_wcpay_feature_upe_settings_preview` so that we can toggle between old settings and the new settings. We also later allowed users to enroll into this "new settings preview" to get some early feedback.

UPE and the "new settings UI" should be the default now as you mentioned. The additional UPE methods can only be accessed via our new settings UI when UPE is enabled.

I think the `isUpeSettingsPreviewEnabled` handling here is probably not needed. We can tackle this in another PR though, I don't think it blocks this from merging.